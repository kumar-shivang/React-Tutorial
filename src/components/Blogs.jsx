import { useLocation } from "react-router-dom";

export default function Blogs() {
  const location = useLocation();
  const state = location.state;
  console.log(state);
  return (
    <div>
      <h1>Blogs</h1>
      {state ? (
        <div>
          <h2>{state.blogs[0].title}</h2>
          <p>{state.blogs[0].content}</p>
        </div>
      ) : (
        <h2>No blog found</h2>
      )}
    </div>
  );
}
