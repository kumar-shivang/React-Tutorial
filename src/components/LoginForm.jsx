export default function LoginForm() {
  return (
    <form>
      <label htmlFor="username">Username</label>
      <input type="text" id="username" />
      <label htmlFor="password">Password</label>
      <input type="password" id="password" />
      <button type="submit">Login</button>
    </form>
  );
}
