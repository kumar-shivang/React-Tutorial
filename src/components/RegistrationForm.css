form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 20px;
}

input {
  margin: 10px;
  padding: 10px;
  width: 300px;
  border-radius: 5px;
  border: 1px solid #ccc;
}

label {
  margin: 10px;
}

button {
  margin: 10px;
  padding: 10px;
  width: 300px;
  border-radius: 5px;
  border: 1px solid #ccc;
  background-color: #f0f0f0;
  cursor: pointer;
}

button:hover {
  background-color: #e0e0e0;
}
