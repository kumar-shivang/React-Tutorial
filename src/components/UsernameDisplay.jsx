export function UsernameDisplay({ username }) {
  return (
    <>
      <b>
        <h1>{username}</h1>
      </b>
      <br />
    </>
  );
}
