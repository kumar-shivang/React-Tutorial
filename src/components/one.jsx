import Two from "./two";
export default function One() {
  return (
    <div id="one">
      <h1>One</h1>
      <Two />
    </div>
  );
}
