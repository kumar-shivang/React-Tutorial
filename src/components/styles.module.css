.username {
  font-weight: bold;
  font-size: 1.25rem;
  color: darkred;
}
