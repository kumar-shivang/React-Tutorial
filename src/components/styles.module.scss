.food {
  font-size: 2rem;
  color: darkblue;
}
