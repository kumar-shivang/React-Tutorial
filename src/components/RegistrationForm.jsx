import "./RegistrationForm.css";
import { useState } from "react";
export default function RegistrationForm() {
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const isDisabled = username === "" || email === "" || password === "";
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        let formData = new FormData(e.target);
        console.log(formData.get("username"));
        console.log(formData.get("email"));
        console.log(formData.get("password"));
        console.log("Form Submitted");
      }}
    >
      <label htmlFor="username">Username</label>
      <input
        type="text"
        id="username"
        name="username"
        value={username}
        onChange={(e) => setUsername(e.target.value)}
      />
      <br />
      <label htmlFor="email">Email</label>
      <input
        type="email"
        id="email"
        name="email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <br />
      <label htmlFor="password">Password</label>
      <input
        type="password"
        id="password"
        name="password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <br />
      <button type="submit" disabled={isDisabled}>
        Register
      </button>
    </form>
  );
}
