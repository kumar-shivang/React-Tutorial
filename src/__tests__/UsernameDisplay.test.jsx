import { describe, it, expect } from "vitest";
import { UsernameDisplay } from "../components/UsernameDisplay";
import { render } from "@testing-library/react";
describe("UsernameDisplay", () => {
  it("should render username", () => {
    const result = render(<UsernameDisplay username={"user1"} />);
    expect(result.container).toMatchSnapshot();
  });
});
