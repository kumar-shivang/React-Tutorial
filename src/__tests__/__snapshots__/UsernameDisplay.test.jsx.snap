// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsernameDisplay > should render username 1`] = `
<div>
  <b>
    <span>
      user1
    </span>
  </b>
  <br />
</div>
`;
